packages/flutter/test/widgets/binding_first_frame_rasterized_test.dart

```diff
@@ -0,0 +1,39 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:ui';
+
+import 'package:flutter/material.dart';
+import 'package:flutter_test/flutter_test.dart';
+import 'package:quiver/testing/async.dart';
+
+void main() {
+  setUp(() {
+    WidgetsFlutterBinding.ensureInitialized();
+  });
+
+  test('Deferred frames will trigger the first frame callback', () {
+    FakeAsync().run((FakeAsync fakeAsync) {
+      final WidgetsBinding binding = WidgetsBinding.instance;
+      binding.deferFirstFrame();
+
+      runApp(const Placeholder());
+      fakeAsync.flushTimers();
+
+      // Simulates the engine completing a frame render to trigger the
+      // appropriate callback setting [WidgetBinding.firstFrameRasterized].
+      binding.window.onReportTimings(
```

Comment:
Please let me know if there's a better way to do this, as there doesn't seem to be an engine in the tests that actually calls the callbacks.